Neos.Flow/Classes/Mvc/ActionResponse.php

```diff
@@ -231,13 +227,36 @@ public function getContentType(): string
         return $this->contentType ?? '';
     }
 
-    /**
-     * Use this if you want build your own HTTP Response inside your action
-     * @param ResponseInterface $response
-     */
-    public function replaceHttpResponse(ResponseInterface $response): void
+    public function replaceHttpResponse(ResponseInterface $httpResponse): void
     {
-        $this->httpResponse = $response;
+        $this->reset();
+
+        $this->setStatusCode($httpResponse->getStatusCode());
+        $this->setContent($httpResponse->getBody());
+
+        $contentType = $httpResponse->getHeader('Content-Type');
+        if (!empty($contentType)) {
+            $this->setContentType(reset($contentType));
+            $httpResponse = $httpResponse->withoutHeader('Content-Type');
+        }
+
+        $location = $httpResponse->getHeader('Location');
+        if (!empty($location)) {
+            $this->setRedirectUri(new Uri(reset($location)), $httpResponse->getStatusCode());
+            $httpResponse = $httpResponse->withoutHeader('Location');
+        } else {
+            $this->setStatusCode($httpResponse->getStatusCode());
+        }
+
+        foreach ($httpResponse->getHeader('Set-Cookie') as $value) {
+            $cookie = Cookie::createFromRawSetCookieHeader($value);
```

Comment:
not sure if its wise to introduce there the first actual usage: https://github.com/neos/flow-development-collection/issues/3288

Reply:
I would mid term want to get rid of the whole Cookie object or remodel it maybe? IDK, sure a DTO to make getCookieParams/withCookieParams a bit nicer to use would be great, but right now the cookie object stems from our own http implementation way back. So anything we can cut from it is a win IMHO 😂